packages/std/src/math.rs

```diff
@@ -0,0 +1,255 @@
+use schemars::JsonSchema;
+use serde::{de, ser, Deserialize, Deserializer, Serialize};
+use std::convert::{TryFrom, TryInto};
+use std::{fmt, ops};
+
+use crate::errors::{generic_err, underflow, StdError, StdResult};
+
+/// Decimal9 represents a fixed-point decimal value with 9 fractional digits.
+/// That is Decimal9(1_000_000_000) == 1
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Decimal9(pub u64);
+
+pub const DECIMAL_FRACTIONAL: u64 = 1_000_000_000;
+
+impl Decimal9 {
+    pub fn one() -> Decimal9 {
+        Decimal9(DECIMAL_FRACTIONAL)
+    }
+
+    // convert integer % into Billionth units
+    pub fn percent(percent: u64) -> Decimal9 {
+        Decimal9(percent * 10_000_000)
+    }
+
+    // convert permille (1/1000) into Billionth units
+    pub fn permille(permille: u64) -> Decimal9 {
+        Decimal9(permille * 1_000_000)
+    }
+}
+
+//*** Uint128 ***/
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, JsonSchema)]
+pub struct Uint128(#[schemars(with = "String")] pub u128);
+
+impl Uint128 {
+    pub fn u128(&self) -> u128 {
```

Comment:
If we have this getter function, shouldn't `self.0` be private then? Or do we need it for serialization?

Reply:
It was nice for constructing values. `Uint128(17)` rather than `Uint128::from(17)`. And can't really hurt, can it?

Reply:
Let's just say if we don't have to make it private to guarantee the type-safety, I'd rather make it public.

Reply:
I agree. This case is less critical than Decimal9